### PR TITLE
Major Music Functionality Addition

### PR DESCRIPTION
**Pull request by F1shOutaWater**
*Music Functionality almost entirely there(Add, swap, remove, progress bar, etc)
*Update to GUI look plus additional buttons (honing in on final look and feel)

*TODO: 
*Progress bar has minor bug where on multiple tracks, the thread wants to die, or improper total time duration. 
*Also, AIFF tracks are proving very difficult to accomodate. They are NOT supported at this time.

### CODE REVIEW

**Review by TheDeltaAgent — CHANGES_REQUESTED** [review]
It says the only changes here are binary files when I imagine some actual stuff was changed. I really don't trust this. There are no changes to the form or class at all? Also do you have the class name change (SlideHandler to SlideShowHandler?)

File: Slide-by-Side_creator/creator/SlideHandler.cs [hunk 1]
```diff
@@ -3,50 +3,76 @@
 using System.Linq;
 using System.Text;
 using System.Threading.Tasks;
+using System.Windows.Media;
+using System.Threading;
 
 namespace formNamespace
 {
-    class SlideHandler
+    public struct Slide
```
Comment (yingren-wang) [inline Slide-by-Side_creator/creator/SlideHandler.cs:11]: I thought we discussed about having both structs Slide and Soundtrack as variables under the class SlideShowHandler, so please put them back under the class SlideShowHandler. Correct me if I got it wrong, or please say the reason if you think this is a better choice.

File: Slide-by-Side_creator/creator/SlideHandler.cs [hunk 2]
```diff
@@ -89,8 +115,62 @@ public void createSlide(string path)
             slide.Duration = 5;
         }
 
+<<<<<<< HEAD
```
Comment (yingren-wang) [inline Slide-by-Side_creator/creator/SlideHandler.cs:118]: unresolved github merge reminder, please delete this.

File: Slide-by-Side_creator/creator/SlideHandler.cs [hunk 3]
```diff
@@ -3,50 +3,76 @@
...
-                crossFade,
-                none
-            }
-        }
-        public Slide slide;
+        public MediaPlayer soundTrackPlayer = new MediaPlayer();
+
+        private readonly object _lock = new object();
+        //List to hold all slides
+        private Slide[] slideList;
```
Comment (yingren-wang) [inline Slide-by-Side_creator/creator/SlideHandler.cs:58]: Wrong type of slideList, I think this is a previous mistake we had and it just happens to be here, please delete line 57 and line 58.

**Comment by yingren-wang** [comment]
Please change the pull request from merging into branch **slide-handler-class-branch** to branch **new-master**, since branch **new-master** has the merged change of:
1. Chandler's SlideShowHandler class (was named as SlideHandler)
2. Yingren's add slide to slideshow panel
3. Ethan's dropdown menu on picture box

So in branch **new-master**, it can add slide to slideshow panel by using Ethan's dropdown menu on the picture box. But the information like the slideList is not stored in the class SlideShowHandler yet.

**Review by yingren-wang — CHANGES_REQUESTED** [review]

**Comment by yingren-wang** [comment]
Please delete all the following files from the pull request:
(You can choose to save them locally, but please do not push them to the github repo)
1. .vs/
2. */Debug/ (including obj/Debug/.. and bin/Debug/..)
Thx

File: Slide-by-Side_creator/creator/Form1.cs [hunk 4]
```diff
@@ -30,15 +33,39 @@ public partial class Form1 : Form
         private System.Windows.Forms.Timer time = new System.Windows.Forms.Timer();
```
Comment (yingren-wang) [inline Slide-by-Side_creator/creator/Form1.cs:33]: Is this variable the same as the private bool musicPlaying variable?
Please check line 39.

**Review by yingren-wang — CHANGES_REQUESTED** [review]

File: Slide-by-Side_creator/creator/SlideHandler.cs (hunk 1 above)
Comment (F1shOutaWater) [reply Slide-by-Side_creator/creator/SlideHandler.cs:11]: We're going to put it outside :)

**Pull request closed without merging**